metadata.rb

```diff
@@ -1,48 +1,14 @@
-name              'memcached'
-maintainer        'Chef Software, Inc.'
-maintainer_email  'cookbooks@chef.io'
-license           'Apache 2.0'
-description       'Installs memcached and provides a define to set up an instance of memcache via runit'
-long_description  IO.read(File.join(File.dirname(__FILE__), 'README.md'))
-version           '1.8.0'
-
-depends           'runit', '~> 1.0'
-depends           'yum', '~> 3.0'
-depends           'yum-epel'
-
-supports 'amazon'
-supports 'centos'
-supports 'debian'
-supports 'fedora'
-supports 'redhat'
-supports 'scientific'
-supports 'smartos'
-supports 'suse'
-supports 'ubuntu'
-
-recipe 'memcached', 'Installs and configures memcached'
-
-attribute 'memcached/memory',
-          :display_name => 'Memcached Memory',
-          :description  => 'Memory allocated for memcached instance',
-          :default      => '64'
-
-attribute 'memcached/port',
-          :display_name => 'Memcached Port',
-          :description  => 'Port to use for memcached instance',
-          :default      => '11211'
-
-attribute 'memcached/user',
-          :display_name => 'Memcached User',
-          :description  => 'User to run memcached instance as',
-          :default      => 'nobody'
-
-attribute 'memcached/listen',
-          :display_name => 'Memcached IP Address',
-          :description  => 'IP address to use for memcached instance',
-          :default      => '0.0.0.0'
-
-attribute 'memcached/logfilename',
-          :display_name => 'Memcached logfilename',
-          :description  => 'The filename used to log memcached',
-          :default      => 'memcached.log'
```

Comment:
Attributes in the metadata are a bit of an anacronism..

They're not consumed by anything,they require tedious duplication between
the metadata.rb file and the attributes files, and become inaccurate if
there is any kind of load-time decision logic happening

-s

On Thu, Sep 3, 2015 at 2:51 PM, Nathen Harvey <notifications@github.com>
wrote:

> In metadata.rb
> <https://github.com/opscode-cookbooks/memcached/pull/48#discussion_r38641654>
> :
>
> > -          :default      => '11211'
> > -
> > -attribute 'memcached/user',
> > -          :display_name => 'Memcached User',
> > -          :description  => 'User to run memcached instance as',
> > -          :default      => 'nobody'
> > -
> > -attribute 'memcached/listen',
> > -          :display_name => 'Memcached IP Address',
> > -          :description  => 'IP address to use for memcached instance',
> > -          :default      => '0.0.0.0'
> > -
> > -attribute 'memcached/logfilename',
> > -          :display_name => 'Memcached logfilename',
> > -          :description  => 'The filename used to log memcached',
> > -          :default      => 'memcached.log'
>
> Why not have supported attributes listed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opscode-cookbooks/memcached/pull/48/files#r38641654>.
>